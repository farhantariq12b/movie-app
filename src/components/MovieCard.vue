<template>
  <div class="max-w-md h-md rounded overflow-hidden shadow-lg m-2">
    <img class="w-full" src="https://random.imagecdn.app/500/150" alt="Sunset in the mountains">
    <div class="px-6 py-4">
      <div data-testid="title" class="font-bold text-xl mb-2">{{ movie.Title }}</div>
    </div>
    <div data-testid=body class="px-2 pt-4 pb-2 flex flex-row justify-between">
      <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">ID: {{movie.imdbID}}</span>
      <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">Year: {{movie.Year}}</span>
      <span @click="toggleFavourite(movie.imdbID)" :class="[movie.favourite ? 'bg-yellow-200': 'bg-gray-200']" class="inline-block pointer rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mb-2">Favourite</span>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import Movie from '@/types/Movie';

export default defineComponent({
  name: 'MovieCard',
  props: {
    movie: {
      required: true,
      type: Object as PropType<Movie>
    }
  },
  setup (props, { emit }) {
    function toggleFavourite (imdbID: string) {
      emit('toggleFavourite', imdbID)
    }
    return {
      toggleFavourite
    }
  }
})

</script>
<style scoped lang="scss">
  .pointer {
    cursor: pointer;
  }
</style>
