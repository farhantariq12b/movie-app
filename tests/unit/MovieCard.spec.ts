import { mount, shallowMount } from "@vue/test-utils";
import MovieCard from "@/components/MovieCard.vue";

describe("MovieCard.vue", () => {
  it("should render HTML correctly", () => {
    const movie = {
      Title: 'Water',
      Year: '2021',
      imdbID: '1234',
      favourite: true
    };
    const wrapper = shallowMount(MovieCard, {
      props: { movie },
    });
		expect(wrapper.vm.$el).toMatchSnapshot();
	});

  it("renders movie props correctly when passed", async () => {
    const movie = {
      Title: 'Water',
      Year: '2021',
      imdbID: '1234',
      favourite: true
    };
    const wrapper = shallowMount(MovieCard, {
      props: { movie },
    });
    const result = ['ID: 1234', 'Year: 2021', 'Favourite']
    let title = wrapper.find('[data-testid=title]');
		expect(title.text()).toBe("Water")
    let body = wrapper.findAllComponents('[data-testid=body] span');
    body.forEach((span, index) => {
      expect(span.text()).toBe(result[index])
    })
  });

});
