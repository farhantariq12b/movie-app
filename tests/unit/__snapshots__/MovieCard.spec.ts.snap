// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard.vue should render HTML correctly 1`] = `
<div
  class="max-w-md h-md rounded overflow-hidden shadow-lg m-2"
>
  <img
    alt="Sunset in the mountains"
    class="w-full"
    src="https://random.imagecdn.app/500/150"
  />
  <div
    class="px-6 py-4"
  >
    <div
      class="font-bold text-xl mb-2"
      data-testid="title"
    >
      Water
    </div>
  </div>
  <div
    class="px-2 pt-4 pb-2 flex flex-row justify-between"
    data-testid="body"
  >
    <span
      class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
    >
      ID: 1234
    </span>
    <span
      class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
    >
      Year: 2021
    </span>
    <span
      class="bg-yellow-200 inline-block pointer rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mb-2"
    >
      Favourite
    </span>
  </div>
</div>
`;
